components/textfield/stories/textfield.stories.js

```diff
@@ -122,24 +141,133 @@ export default {
-export const TextArea = Template.bind({});
-TextArea.tags = ["!dev"];
-TextArea.args = {
-	labelText: "Username",
-	multiline: true,
-	grows: true,
-	value: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.",
+
+/**
+ * Text fields can display a character count indicator when the length of the text entry needs to be kept under a predefined value. Character count indicators can be used in conjunction with other indicators (validation icon, “optional” or “required” indicators) when necessary.
+*/
+export const CharacterCount = Template.bind({});
+CharacterCount.tags = ["!dev"];
+CharacterCount.args = {
+	hasCharacterCount: true,
+	characterCount: 24,
+	value: "lisawilson23"
+};
+CharacterCount.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+/**
+ * A text field in a disabled state shows that an input field exists, but is not available in that circumstance. This can be used to maintain layout continuity and communicate that a field may become available later.
+*/
+export const Disabled = Template.bind({});
+Disabled.tags = ["!dev"];
+Disabled.args = {
+	isDisabled: true
 };
-TextArea.parameters = {
-	chromatic: { disableSnapshot: true },
+Disabled.parameters = {
+	chromatic: { disableSnapshot: true }
 };
 
+
+/**
+ * A text field can be marked as having an error to show that a value needs to be entered in order to move forward or that a value that was entered is invalid. If an error exists, the error icon always overrides the validation icon.
+*/
+export const Invalid = Template.bind({});
+Invalid.tags = ["!dev"];
+Invalid.args = {
+	isInvalid: true
+};
+Invalid.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+/**
+ * A text field can have [help text](/docs/components-help-text--docs) below the field to give extra context or instruction about what a user should input in the field. The help text area has two options: a description and an error message. The description communicates a hint or helpful information, such as specific requirements for correctly filling out the field. The error message communicates an error for when the field requirements aren’t met, prompting a user to adjust what they had originally input.
+ *
+ * Instead of placeholder text, use the help text description to convey requirements or to show any formatting examples that would help user comprehension. Putting instructions for how to complete an input, requirements, or any other essential information into placeholder text is not accessible.
+*/
+export const HelpText = HelpTextOptions.bind({});
+HelpText.tags = ["!dev"];
+HelpText.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+
+export const Quiet = QuietGroup.bind({});
```

Comment:
LOVE this! What would you think about just adding some of the extra quiet guidelines for the description? It's not necessary, but it'd be nice for users I think.

https://spectrum.adobe.com/page/text-field/#Quiet